File: Enhancement One Original/projectpackage/Task.java
package projectpackage;

public class Task {
	
	private String taskID;
	private String taskName;
	private String taskDesc;
	
	//constructor
	public Task(String id, String name, String description) {
		if (id == null || id.length() > 10) {
			throw new IllegalArgumentException("Invalid ID");
		}
		if (name == null || name.length() > 20) {
			throw new IllegalArgumentException("Invalid Name");
		}
		if (description == null || description.length() > 50) {
			throw new IllegalArgumentException("Invalid Description");
		}
		
		taskID = id;
		taskName = name;
		taskDesc = description;
	}
	
	//getters
	public String GetID() {
		return taskID;
	}
	public String GetTaskName() {
		return taskName;
	}
	public String GetTaskDescription() {
		return taskDesc;
	}
	
	//setters
	public void SetTaskID(String id) {
		if (id == null || id.length() > 10) {
			throw new IllegalArgumentException("Invalid ID");
		}
		taskID = id;
		}
	public void SetTaskName(String name) {
		if (name == null || name.length() > 20) {
			throw new IllegalArgumentException("Invalid Name");
		}
		taskName = name;
		}
	public void SetTaskDesc(String description) {
		if (description == null || description.length() > 50) {
			throw new IllegalArgumentException("Invalid Description");
		}
		taskDesc = description;
		}
	}

File: Enhancement One New/Task.py
import TaskService
import Appointment
import AppointmentService
import Contact
import ContactService

class Task:
    def __init__(self, task_id, task_name, task_desc):
        if task_id is None or len(task_id) > 10:
            raise ValueError("Invalid ID")
        if task_name is None or len(task_name) > 20:
            raise ValueError("Invalid Name")
        if task_desc is None or len(task_desc) > 50:
            raise ValueError("Invalid Description")

        self.task_id = task_id
        self.task_name = task_name
        self.task_desc = task_desc

    # getters
    def get_id(self):
        return self.task_id

    def get_task_name(self):
        return self.task_name

    def get_task_description(self):
        return self.task_desc

    # setters
    def set_task_id(self, task_id):
        if task_id is None or len(task_id) > 10:
            raise ValueError("Invalid ID")
        self.task_id = task_id

    def set_task_name(self, task_name):
        if task_name is None or len(task_name) > 20:
            raise ValueError("Invalid Name")
        self.task_name = task_name

    def set_task_desc(self, task_desc):
        if task_desc is None or len(task_desc) > 50:
            raise ValueError("Invalid Description")
        self.task_desc = task_desc
